### PR TITLE
Upgrade OpenTelemetry versions, add explicit OkHttp3 artifact -- so w…

### PR DESCRIPTION
**Pull request by alesj**
…e don't grab a random one from dependencies.

Signed-off-by: Ales Justin <ales.justin@gmail.com>

### CODE REVIEW

File: pom.xml [hunk 1]
```diff
@@ -27,13 +27,14 @@
...
+        <kafka-kubernetes-config-provider.version>1.0.0</kafka-kubernetes-config-provider.version>
+        <okhttp3.version>3.14.9</okhttp3.version>
         <jaeger.version>1.6.0</jaeger.version>
         <opentracing.version>0.33.0</opentracing.version>
         <opentracing-kafka-client.version>0.1.15</opentracing-kafka-client.version>
-        <opentelemetry.version>1.7.0-alpha</opentelemetry.version>
-        <opentelemetry-stable.version>1.7.0</opentelemetry-stable.version>
-        <grpc.version>1.41.0</grpc.version>
+        <opentelemetry.version>1.9.0-alpha</opentelemetry.version>
+        <opentelemetry-stable.version>1.9.0</opentelemetry-stable.version>
```
Comment (im-konge) [inline pom.xml:36]: I know that @Frawless asked about it, but why do we need both stable and "not stable" version?

Comment (alesj) [reply pom.xml:36]: B/c OpenTelemetry has this strange way of versioning -- some of the stuff is still not released under non-alpha version:
* e.g. https://search.maven.org/artifact/io.opentelemetry/opentelemetry-sdk-extension-autoconfigure

And we depend on them -- this e.g. makes our OTel usage generic;
we can switch actual tracing impl - Jaeger vs Zipkin vs OTLP vs ...

Comment (im-konge) [reply pom.xml:36]: Okay, thanks for the explanation (again 😄)

**Review by im-konge — APPROVED** [review]
LGTM, assuming it works.

**Comment by im-konge** [comment]
Out of curiosity, does this work with all type of clients in all of our tests? Or did you tried just tracing?

**Comment by im-konge** [comment]
You should try all the clients with the options, because you are hacking the version of OkHttp. It's better to test it properly (few hours) than fixing the issues later.

**Comment by alesj** [comment]
Before it was random OkHttp version, and it worked.

Anyway, how do I try all client tests?

**Comment by im-konge** [comment]
It can use something from newer/older version of OkHttp and it will throw exceptions 🤷 

There are, as always, many way how to test it. You can f.e. deploy Kafka cluster with custom configuration and play with that a bit. Or you can build the images and run the tests and always also check logs, if there is some exception :)

**Comment by alesj** [comment]
Don't we have a CI to do this?
Or how do you test any new stuff?

**Comment by im-konge** [comment]
`Don't we have a CI to do this?`
> Yes, in operators repo. 

` Or how do you test any new stuff?`
> I don't need to repeat myself, it's in the previous comment

**Comment by alesj** [comment]
Geez, manual stuff or good faith on this … nice 🤪

**Comment by im-konge** [comment]
@alesj so you are testing the new features how? Code and pray? You merge it to master and then, after tester writes the tests and he/she find, that it doesn't work, you are fixing it? Or what is your workflow?

**Comment by im-konge** [comment]
If you want to have tests in this repository, you can add them. It wasn't the main thing to do until, but it is planned to be done soon.

**Comment by alesj** [comment]
It looks like it was pray before?
Or else we wouldn’t have multiple OkHttp3 artifacts, and a random would be
used.
And everything gets merged …

**Comment by alesj** [comment]
That’s the problem I guess …

**Comment by alesj** [comment]
> If you want to have tests in this repository, you can add them

OK, let me add some tests then ...

**Comment by im-konge** [comment]
You don't have to add tests, I just wanted from you to check, if this change will not break anything

**Comment by alesj** [comment]
No problem to add some tests.
It should be a simple usage of testcontainers.

**Comment by im-konge** [comment]
That was my idea too.

**Comment by alesj** [comment]
Will split this into 2 PRs.

**Pull request closed without merging**